README.md

```diff
@@ -14,33 +14,156 @@ The system uses a number of open source projects to work properly:
 * [ElizaBOT-JS] - Javascript implementation of the ELIZA specification by Weizenbaum, 1966. Special thanks to Landsteiner!
 
 
-# Installation:
+## Installation:
 
-TEXA requires [Go Lang](https://golang.org/)  v1.7+ to run.
+### Go:
 
-Install the dependencies, devDependencies and start the server.
+TEXA is written in [Go](http://golang.org) v1.7+. If you don't have a Go
+development environment, please [set one up](http://golang.org/doc/code.html).
+
+### Redis: 
+
+Texa using the [texalib]
+to store the slabs, CatPages and MtsPages, in [redis](https://redis.io/)
+[Redis quick start](https://redis.io/topics/quickstart#redis-quick-start).
+
+### MongoDB:
+
+TEXA using [MongoDB](https://docs.mongodb.com/manual/installation) to persist the conversation between 
+AI and human for every session.
+
+### Build on a local OS:
+
+Make sure your go version is 1.7+
 
 ``$ go version``
 
+Move to your 
+
 ``$ cd $GOPATH/src/``
 
-``$ git clone https://github.com/TexaProject/texa.git``
+Clone the texa project from github
 
-``$ go get -u https://github.com/TexaProject/texajson.git``
+``$ git clone https://github.com/TexaProject/texa.git``
 
-``$ go get -u https://github.com/TexaProject/texalib.git``
+Navigate to the texa project
 
 ``$ cd $GOPATH/src/texa/``
 
-``$ go run main.go``
+Run make file, by the following order
+
+```sh
+   make install
+   make test
+   make run
+
+   make fmt
```

Comment:
Can we have this command & the following commands below abstracted into a separate section called CONTRIBUTE ?

Reply:
ok